core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -470,18 +470,27 @@ public void setChannel(@Nonnull ChannelBuilder cb,
 
     /**
      * Shows {@link Channel#classLoadingCount}.
+     * @return Requested value or {@code -1} if the agent is offline.
      * @since 1.495
      */
+    @CheckReturnValue
     public int getClassLoadingCount() throws IOException, InterruptedException {
+        if (getChannel() == null) {
```

Comment:
The main difference i see between the two PRs is that this PR leverages the use of `getChannel()` which only returns a `channel` object. Just a few lines later the same function uses the `channel` object directly. From a readability standpoint this is pretty confusing as a reader might misinterpret that `getChannel()` does something different behind the scenes which in actual fact it doesn't.

Reply:
I did that way because the actual implementati of `getEnvVarsFull` does that way too.